linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2969,13 +2977,17 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
           })
 
         case JSGlobalRef(name) =>
-          js.VarRef(transformGlobalVarIdent(name))
+          if (name == JSGlobalRef.FileLevelThis)
+            globalVar(VarField.fileLevelThis, CoreVar)
+          else
+            js.VarRef(transformGlobalVarIdent(name))
 
         case JSTypeOfGlobalRef(globalRef) =>
           js.UnaryOp(JSUnaryOp.typeof, transformExprNoChar(globalRef))
 
         case JSLinkingInfo() =>
-          globalVar(VarField.linkingInfo, CoreVar)
+          throw new IllegalArgumentException(
+            "JSLinkingInfo is deprecated as of 1.17.0.")
```

Comment:
```suggestion
          throw new IllegalArgumentException(
              "JSLinkingInfo was removed in 1.17.0.")
```
Deprecated suggests there is still a valid implementation of it. Since there isn't, it is confusing to talk about "deprecated".